theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer.tsx

```diff
@@ -250,9 +251,11 @@ export class TraceViewerWidget extends ReactWidget {
         }
     }
 
-    onCloseRequest(msg: Message): void {
+    async onCloseRequest(msg: Message): Promise<void> {
         this.statusBar.removeElement('time-selection-range');
         super.onCloseRequest(msg);
+        const remoteExperiments = await this.experimentManager.getOpenedExperiments();
+        signalManager().fireOpenedTracesChangedSignal(new OpenedTracesUpdatedSignalPayload(remoteExperiments ? remoteExperiments.length : 0));
```

Comment:
Although it works, it's a bit misleading to broadcast this signal here because the opened traces (number of traces in opened traces widget) is not changed by closing a trace viewer tab.

Perhaps it would be better to use CLOSE_TRACEVIEWERTAB signal?

Reply:
That was my first guess as well, but I'm not sure how to make it work
- CLOSE_TRACEVIEWERTAB signal is fired by the opened-traces view's context menu and is caught by trace-viewer.tsx to close the tab. 
- The handler for CLOSE_TRACEVIEWERTAB signal in trace-viewer.tsx calls shell.closeWidget to close the tab. But this handler does not catch the event where the user closes the tab using the tab's remove ('x') button
- Widget.onCloseRequest is called whenever a tab is closed, either from the context menu or the tab's remove button. So I wanted to emit an event in this function, but I don't know if I can emit CLOSE_TRACEVIEWERTAB given that trace-viewer.tsx is watching for the same signal.

Reply:
Yes, that was my observation too. I tried emitting the CLOSE_TRACEVIEWERTAB event and it didn't seem to cause any problem. But OK, we can revisit it later if needed. I think there's a few other issues with how signals are used in the application.